setup.py

```diff
@@ -91,7 +91,9 @@
         "Documentation": "https://docs.wagtail.org",
         "Source": "https://github.com/wagtail/wagtail",
     },
-    packages=find_packages(),
+    packages=find_packages(
+        exclude=["*tests", "*tests*", "tests*", "*tests.*", "tests"],
```

Comment:
This will exclude the `wagtail.tests.utils` sub-package.

IMO trying to both keep and exclude parts of the same package is very fragile.  One typical layout is to move tests out of the package, so that you have two distinct trees.  If the project doesn’t want to do that (understandable with git’s and github’s poor support for file moves), maybe you should rename the utils to `wagtail.testutils` so that you have separate sub-trees for test and test_utils and stop trying to exclude but include the same namespace.

Reply:
It did exclude them. But adding graft wagtail/tests/utils in Manifest.in resolved it.

Reply:
I see. So I should move utils out of the tests directory and rename it?

Reply:
an example of `check-manifest` usage with pre-commit - https://github.com/torchbox/wagtailmedia/blob/main/.pre-commit-config.yaml#L35-L39

Reply:
Thanks, @zerolab. I'll check it out.

Reply:
@zerolab I used check-manifest for both the modified Manifest.in and the original one. I am getting the same error:
Ubuntu VM, Node 16.
![image](https://user-images.githubusercontent.com/86092410/160100483-9da9a912-f6ab-4a0f-aee9-366c86eb79e7.png)

These lines cause the trailing slash error:
::

    prune **/__pycache__
    prune **/static_src
I don't understand the cause of other errors as webpack is installed and have been used to build the project.

Reply:
Also, after the implementation of #8011, 'tests' have been renamed to 'test.' There are two directories, wagtail/test containing and wagtail/tests both of them have 'utils' present in them. Can you tell me why that is?

Reply:
This was because we moved the `wagtail.core` app to the top level so `wagtail.core.tests` clashed with `wagtail.tests`

I thought that `wagtail.test` would be a better name for the module that contains test infrastructure as it doesn't contain any tests and is analogous to Django's `django.test` module.

The `wagtail.tests` contains tests for the core parts of Wagtail. I didn't merge this into `wagtail.test` as that contains test infrastructure used by the whole of Wagtail.

We should not exclude `wagtail.test` as this contains test infrastructure that can be reused by third-party packages. It includes the `wagtail.tests.utils` module (now `wagtail.test.utils`). And for backwards compatibility, we should not exclude any of the following modules that redirect imports to `wagtail.test`:
- `wagtail/tests/utils/*`
- `wagtail/tests/dummy_external_storage.py`
- `wagtail/tests/settings.py`

Reply:
As I understand the implications of the `tests` -> `test split seem excellent for us to be able to exclude the right thing?

- `wagtail/test` should all be included
- `wagtail/tests` should all be excluded, once we no longer have to keep redirect imports in there for backwards compatibility?

@kaedroho at what point do you think we will be able to remove those re-exports?

Reply:
As I understand the implications of the `tests` -> `test split seem excellent for us to be able to exclude the right thing?

- `wagtail/test` should all be included
- `wagtail/tests` should all be excluded, once we no longer have to keep redirect imports in there for backwards compatibility?

@kaedroho at what point do you think we will be able to remove those re-exports?

Reply:
fwiw, we will move to using flit for packaging on wagtail-localize, and it looks like has better support for explicit exclusions ([example](https://github.com/wagtail/wagtail-localize/pull/589/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R66)) and best of all, that is respected when building wheels

Reply:
Have tested this and it also works. Looks like with distutils being deprecated, setuptools have finally picked up the slack and no longer ignore the `exclude` clause when building wheels \o/

Reply:
Following up – As of #9987, the confirmed deletion timeline for `wagtail.tests.utils` is v6.0. Then we’ll be able to exclude `tests` completely.